src/lib/summary.ts

```diff
@@ -2,51 +2,127 @@ import fs from "fs";
 import { Matches } from "./comparison";
 import { Range } from "./range";
 export type RangesTuple = [Range, Range];
+
+/**
+ * @param minimumLinesInLargestFragment The minimum amount of lines required by the largest code fragment. The default
+ * is 1.
+ * @param minimumLinesInSmallestFragment The minimum amount of lines required by the smallest code fragment. The default
+ * is 0.
+ * @param fragmentOutputLimit The maximum amount of code fragments that will appear in the output. Everything is shown
+ * if it is anything other then a number greater then zero.
+ */
+export interface FilterOptions {
+  fragmentOutputLimit?: number;
+  minimumLinesInLargestFragment?: number;
+  minimumLinesInSmallestFragment?: number;
+}
+
 export class Summary {
+  public readonly gapSize: number;
   private readonly results: Map<string, Matches<Range>>;
-  private readonly minimumMaximumLines: number;
-  private readonly gapSize: number;
-  private readonly minimumMinimumLines: number;
+  private readonly filterOptions: FilterOptions;
+
+  private readonly defaultFilterOptions: FilterOptions = {
```

Comment:
If these values aren't meant to be changed, also wrap the right part with `Object.freeze()`